src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -28,81 +28,338 @@
+
+    std::string name() {
+        std::string name = data.group.object.name;
+
+        if(!data.group.name.empty()) {
+            name += ":" + data.group.name;
+        }
+
+        if(data.materialId != -1) {
+            name += "$" + std::to_string(data.materialId);
+        }
+
+        const int numPrimitiveTypes =
+            ((data.points) ? 1 : 0)
+            + ((data.lines) ? 1 : 0)
+            + ((data.faces) ? 1 : 0);
+        if(numPrimitiveTypes > 1) {
+            std::ostringstream out;
+            Debug(&out) << primitive;
+            name += "%" + out.str();
+        }
+
+        return name;
+    }
+};
+
+
+/* The state of the imported generated by openData() */
+struct ImporterState {
+    std::vector<ObjMaterial> materials;
+    std::unordered_map<std::string, int> materialIds;
+
+    std::vector<std::string> textures;
+    std::unordered_map<std::string, int> textureIds;
+
+    std::vector<std::unique_ptr<ObjObject>> objects;
+
+    std::vector<std::unique_ptr<ObjMesh>> meshes;
+    std::unordered_map<std::string, int> meshIds;
+
+    std::vector<std::string> meshlessObjects;
+    std::unordered_map<std::string, int> meshlessObjectIds;
+
+    std::vector<Vector3> positions;
+    std::vector<Vector2> texCoords;
+    std::vector<Vector3> normals;
 };
 
 namespace {
 
-void ignoreLine(std::istream& in) {
-    in.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+int strToInt(const ArrayView<char> str) {
+    char* err;
+    const int i = int(strtol(str.data(), &err, 10));
+    if(err == nullptr) {
+        Error() << "Trade::ObjImporter::mesh3D(): error while converting numeric data";
+        return 0;
+    }
+
+    return i;
 }
 
-template<std::size_t size> Math::Vector<size, Float> extractFloatData(const std::string& str, Float* extra = nullptr) {
-    std::vector<std::string> data = Utility::String::splitWithoutEmptyParts(str, ' ');
-    if(data.size() < size || data.size() > size + (extra ? 1 : 0)) {
-        Error() << "Trade::ObjImporter::mesh3D(): invalid float array size";
-        throw 0;
+float strToFloat(const ArrayView<char> str) {
+    char* err;
+    const float f = strtof(str.data(), &err);
+    if(err == nullptr) {
+        Error() << "Trade::ObjImporter::mesh3D(): error while converting numeric data";
+        return 0;
+    }
+
+    return f;
+}
+
+// find first index of c in pos, cancel search at newline or whitespace if flag set.
+int findNext(const ArrayView<char> pos, char c, bool termByNewline=false, bool termByWhitespace=false) {
+    const int size = pos.size();
+    for(int i = 0; i < size; ++i) {
+        if(pos[i] == c) {
+            return i;
+        }
+
+        if((pos[i] == '\n' && termByNewline) || (pos[i] == ' ' && termByWhitespace)) {
+            return -1;
+        }
+    }
+    return -1;
+}
+
+// find next non-whitespace char and return suffix at that point.
+// param newlineIsWhitespace if "true", '\n' and '\r' are skipped also
```

Comment:
I know! :D It is a temporary comment, so I don't forget to put decent wording for other people there.